x/observer/genesis.go

```diff
@@ -94,16 +94,29 @@ func InitGenesis(ctx sdk.Context, k keeper.Keeper, genState types.GenesisState)
 		k.SetLastObserverCount(ctx, &types.LastObserverCount{LastChangeHeight: 0, Count: observerCount})
 	}
 
+	tss := types.TSS{}
+	genStateTss := false
 	if genState.Tss != nil {
-		tss := *genState.Tss
+		tss = *genState.Tss
+		genStateTss = true
 		k.SetTSS(ctx, tss)
+	}
+
+	// Set all the pending nonces
+	if genState.PendingNonces != nil {
+		for _, pendingNonce := range genState.PendingNonces {
+			k.SetPendingNonces(ctx, pendingNonce)
+		}
+	} else {
 		for _, chain := range common.DefaultChainsList() {
-			k.SetPendingNonces(ctx, types.PendingNonces{
-				NonceLow:  0,
-				NonceHigh: 0,
-				ChainId:   chain.ChainId,
-				Tss:       tss.TssPubkey,
-			})
+			if genStateTss {
```

Comment:
Why not just checking `genState.Tss != nil`?
Otherwise I would also rename the variable to make it more explicit `existingTSSInGenState`

Reply:
745da8ca35cb6e1d6679310c1f979aa9d1ae1eb4